### PR TITLE
openstack: Refactor TFVars generation

### PR DESCRIPTION
**Pull request by pierreprinetti**
This is expected to be pure refactoring without any functional change.

### CODE REVIEW

File: pkg/tfvars/openstack/openstack.go [hunk 1]
```diff
@@ -182,150 +122,130 @@ func tfVars(masterConfigs []*v1alpha1.OpenstackProviderSpec, workerConfigs []*v1
...
+	serviceCatalog, err := getServiceCatalog(cloud)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not retrieve service catalog: %w", err)
 	}
 
-	tokenID, err := getAuthToken(cloud)
-	if err != nil {
-		return nil, err
+	var regionName string
```
Comment (mandre) [inline pkg/tfvars/openstack/openstack.go:137]: You could potentially move this into getBootstrapShim() as it doesn't appear to be used anywhere else. This way, that's one less arg to pass to getBootstrapShim().

File: pkg/tfvars/openstack/openstack.go [hunk 2]
```diff
@@ -100,80 +77,43 @@ func TFVars(
...
-		bootstrapIgn,
-		installConfig.Config.ControlPlane.Platform.OpenStack,
-		workermpool,
-		installConfig.Config.OpenStack.DefaultMachinePlatform,
-		installConfig.Config.Platform.OpenStack.MachinesSubnet,
-		installConfig.Config.Proxy,
-	)
-}
-
-func tfVars(masterConfigs []*v1alpha1.OpenstackProviderSpec, workerConfigs []*v1alpha1.OpenstackProviderSpec, cloud string, externalNetwork string, externalDNS []string, apiFloatingIP string, ingressFloatingIP string, apiVIP string, ingressVIP string, baseImage string, baseImageProperties map[string]string, infraID string, userCA string, bootstrapIgn string, mastermpool, workermpool, defaultmpool *types_openstack.MachinePool, machinesSubnet string, proxy *types.Proxy) ([]byte, error) {
```
Comment (mandre) [inline pkg/tfvars/openstack/openstack.go:144]: Goodbye tfVars(), I'm not gonna miss you.

**Review by mandre — COMMENTED** [review]
Nice, no more of this monstrosity of a function signature.
Just a small suggestion for which I left a hold if you want to follow it, otherwise feel free to remove the hold.

/lgtm
/hold

File: pkg/tfvars/openstack/openstack.go (hunk 1 above)
Comment (pierreprinetti) [reply pkg/tfvars/openstack/openstack.go]: Good idea!

**Review by mandre — COMMENTED** [review]
/lgtm
/approve
/hold for CI

**Comment by EmilienM** [comment]
nice work! Thanks a lot
/lgtm

**Comment by pierreprinetti** [comment]
/hold
Targeting 4.13

**Comment by pierreprinetti** [comment]
/hold cancel
Ready for review/merge!

@mandre can you please take a look? I have rebased and resolved a conflict with https://github.com/openshift/installer/pull/6602

**Comment by pierreprinetti** [comment]
Looks like the CI is on fire.

/test e2e-openstack
/retest-required

**Review by mandre — COMMENTED** [review]
/lgtm

**Pull request merged**